### PR TITLE
feat: avoid some cloning when mirror requests to flownode

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

1. Do not spawn to check `source table` since there already has cache for this.
2. Avoid to cloning all requests to mirror.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/operator/src/insert.rs [hunk 1]
```diff
@@ -238,8 +202,34 @@ impl Inserter {
...
+                    .context(JoinTaskSnafu)
+                {
+                    warn!(err; "Failed to insert data into flownode");
+                }
+            }
+            Err(err) => warn!(err; "Failed to mirror request to flownode"),
+        }
+
+        let write_tasks = self
+            .group_requests_by_peer(requests)
```
Comment (fengjiachun) [inline src/operator/src/insert.rs:232]: Avoid cloning the request

File: src/operator/src/insert.rs [hunk 2]
```diff
@@ -294,7 +286,7 @@ impl Inserter {
 
                     if !peers.is_empty() {
                         let mut reqs = RegionInsertRequests::default();
-                        reqs.requests.push(req);
+                        reqs.requests.push(req.clone());
```
Comment (fengjiachun) [inline src/operator/src/insert.rs:289]: Only clone those that are necessary

File: src/operator/src/insert.rs [hunk 3]
```diff
@@ -310,14 +302,26 @@ impl Inserter {
...
                 inserts
-                    .entry(flownode.clone())
+                    .entry(peers[0].clone())
                     .or_default()
                     .requests
-                    .extend(reqs.requests.clone());
+                    .extend(reqs.requests);
+                continue;
+            } else {
+                // TODO(discord9): need to split requests to multiple flownodes
```
Comment (fengjiachun) [inline src/operator/src/insert.rs:314]: cc @discord9

File: src/operator/src/insert.rs [hunk 4]
```diff
@@ -238,8 +202,34 @@ impl Inserter {
...
 
-        let tasks = self
-            .group_requests_by_peer(requests.clone())
+        // Mirror requests for source table to flownode
+        match self.mirror_flow_node_requests(&requests).await {
+            Ok(flow_requests) => {
+                let node_manager = self.node_manager.clone();
+                let flow_tasks = flow_requests.into_iter().map(|(peer, inserts)| {
+                    let node_manager = node_manager.clone();
+                    common_runtime::spawn_write(async move {
```
Comment (waynexia) [inline src/operator/src/insert.rs:211]: how about `spawn_bg` to avoid impact front-ground write

Comment (fengjiachun) [reply src/operator/src/insert.rs]: Looks good.

File: src/operator/src/insert.rs (hunk 3 above)
Comment (discord9) [reply src/operator/src/insert.rs:314]: make sense since we are only at standalone mode now

File: src/operator/src/insert.rs [hunk 5]
```diff
@@ -238,8 +202,34 @@ impl Inserter {
             ..Default::default()
         });
 
-        let tasks = self
-            .group_requests_by_peer(requests.clone())
+        // Mirror requests for source table to flownode
+        match self.mirror_flow_node_requests(&requests).await {
```
Comment (killme2008) [inline src/operator/src/insert.rs:206]: What about adding a metric here?

**Review by killme2008 — APPROVED** [review]
Good job!

**Review by discord9 — APPROVED** [review]

File: src/operator/src/insert.rs (hunk 5 above)
Comment (fengjiachun) [reply src/operator/src/insert.rs:206]: 1. Added mirror rows count metric,
2. We already have metric in the cache manager

**Pull request merged**